Apply `ormolu` to all Haskell files to ensure consistent formatting.

lehins opened the pull request:
I found it a bit inconvenient that some projects are required to be formatted with ormolu, while others are formatted in different styles. Avoidance of accidental ormolization of files that aren't formatted with ormolu is somewhat of a burden (even if they are rarely changed). Also for the purpose of readability I think it would be beneficial to have the whole repo formatted consistently.

Feel free to close this PR if this is undesired for any reason.

Command that was used for ormolization (PR reviewer is encouraged to verify that changes introduced come only from this command):

```shell
$ git checkout 44c09b3c9a3a01f9ff1c8de3dcd0f0c7c030f49a # current origin/master
$ find . -path ./dist-newstyle -prune -o -type f -path '*.hs' -exec ormolu -c -m inplace {} +
```

One file will fail `-c` check and will result in a benign discrepancy:

```shell
$ ormolu -m inplace byron/ledger/impl/test/Test/Cardano/Chain/Byron/API.hs
$ ormolu -c -m inplace byron/ledger/impl/test/Test/Cardano/Chain/Byron/API.hs
$ git diff origin/lehins/ormolize
```

lehins committed:
Remove redundant imports thus getting rid of compilation warnings

nc6 commented:
I think this makes sense, but let's also add most of the missing paths to the check (`scripts/ormolise.sh`) so CI verifies this!

lehins committed:
Apply `ormolu` to all Haskell files to ensure consistent formatting.

    Command that was used for ormalization:

    ```shell
    $ find . -path ./dist-newstyle -prune -o -type f -path '*.hs' -exec ormolu -c -m inplace {} +
    ```

lehins committed:
Enable ormolization check on CI for all .hs files in the repo

lehins commented:
> let's also add most of the missing paths to the check (scripts/ormolise.sh) so CI verifies this!

@nc6 Done. Instead of being inclusive I think it is better to just apply that script to all Haskell files. There is no harm in doing so, but at least once a new project is added to the repo we won't have to touch that script.
